src/test/java/com/github/martapietka/ppm2pbm/RgbToGrayscaleByAverageConverterTest.java

```diff
@@ -21,7 +19,8 @@ class RgbToGrayscaleByAverageConverterTest {
     void convertRgbToGrayscale(int r, int g, int b, int expectedResult) {
 
         // WHEN
-        int result = RgbToGrayscaleByAverageConverter.convertRgbToGrayscale(r, g, b);
+        RgbToGrayscaleByAverageConverter rgbToGrayscaleByAverageConverter = new RgbToGrayscaleByAverageConverter();
```

Comment:
Tworzenie testowanego obiesktu tak naprawdę nie jest częścią scenariusza testowego, dodatkowo ma miejsce w każdym teście, więc nie powinno być w sekcji _WHEN_. W takiej sytuacji dobrze jest mieć w klasie testowej pole

```java
private RgbToGrayscaleByAverageConverter rgbToGrayscaleByAverageConverter;
```

lub nawet bardziej ogólne, żeby dodatkowo zaznaczyć w teście polimorfizm typu:

```java
private RgbToGrayscaleConverter rgbToGrayscaleConverter;
```

i pod nim, ale nad metodami testowymi, metodę `setUp()` (nie musi się nazywać dokładnie tak, ale jest to nazwa zwyczajowa), ze stosowną adnotacją. Na przykład:

```java
@BeforeEach
void setUp() {
    this.rgbToGrayscaleConverter = new RgbToGrayscaleByAverageConverter();
}
```

Taki kod wykona się przed każdym testem co z jednej strony trochę upraszcza i porządkuje same testy, a z drugiej - gwarantuje, że każdy test ma "świeżą" instancję do testowania (tutaj to ma mniejsze znaczenie, bo ten akurat obiekt nie ma stanu, ale istotny jest porządek i zachowanie pewnej praktyki 😉).

Powyższa uwaga dotyczy też pozostałych analogicznych testów.